src/coreclr/vm/genericdict.cpp

```diff
@@ -843,12 +843,15 @@ Dictionary::PopulateEntry(
             }
             IfFailThrow(ptr.SkipExactlyOne());
 
+            PTR_MethodTable pMT = th.IsByRef() ? th.GetTypeParam().GetMethodTable() : th.GetMethodTable();
```

Comment:
I do not think this is the right fix. It is just going to crash if the method does something with T, e.g. change the method to `private void M() { Console.WriteLine(typeof(T)); }` to see the problem - I expect that it is going to crash even with this fix.

I think the fix should be in the unsafe accessor emitter.

Reply:
Pushed the fix to unsafe accessor emitter to strip the byref modifier. Also updated the PR title and description to reflect it.